pallets/did/src/utils.rs

```diff
@@ -16,17 +16,45 @@
 
 // If you feel like getting in touch with us, you can do so at info@botlabs.org
 
+use crate::{
+	did_details::{DidCreationDetails, DidPublicKey},
+	BalanceOf, Config, KeyIdOf,
+};
 use parity_scale_codec::Encode;
-use sp_runtime::traits::Hash;
+use sp_core::Get;
+use sp_runtime::traits::{Hash, Zero};
 use sp_std::vec::Vec;
 
-use crate::{did_details::DidPublicKey, Config, KeyIdOf};
-
 pub fn calculate_key_id<T: Config>(key: &DidPublicKey) -> KeyIdOf<T> {
 	let hashed_values: Vec<u8> = key.encode();
 	T::Hashing::hash(&hashed_values)
 }
 
+pub fn calculate_deposit<T: Config>(details: &DidCreationDetails<T>) -> BalanceOf<T>
```

Comment:
Because we still create `DidDetails` from the creation details, perhaps it makes more sense to have this function only on `DidDetails` and, for creations, calculate the deposit after craeting the `DidDetails` instance, so we avoid code duplication.

Reply:
TL;DR: I would remove this function from here and only keep it for `DidCreationDetails`.